service/inject_definition.go

```diff
@@ -144,3 +149,10 @@ func (s *Service) defParametersToService(params map[string]*importer.Parameter)
 	}
 	return ps
 }
+
+func (s *Service) generateAlias() string {
+	alias := strings.Replace(s.Name, " ", "-", -1)
+	alias = strings.Trim(alias, " _-.,")
+	alias = strings.ToLower(alias)
+	return alias
```

Comment:
If I run the same service with a modification in the code, they still share the same volume with this alias. This is a breaking change.
The current behavior is that every changes in the code of the service generates a new id and a new volume.
Maybe this alias need to be generated at the end based on the hash of the service (without the alias) like that if the service is deployed 2 times with the same code, the volume will be the same but if there is a modification in the code then the volume will be different

Reply:
This fixes the problem

```diff
diff --git a/service/inject_definition.go b/service/inject_definition.go
index a3a6c888..80e8926e 100644
--- a/service/inject_definition.go
+++ b/service/inject_definition.go
@@ -12,9 +12,6 @@ import (
 func (s *Service) injectDefinition(def *importer.ServiceDefinition) {
 	s.Name = def.Name
 	s.Alias = def.Alias
-	if s.Alias == "" {
-		s.Alias = s.generateAlias()
-	}
 	s.Description = def.Description
 	s.Repository = def.Repository
 	s.Events = s.defEventsToService(def.Events)
@@ -154,5 +151,5 @@ func (s *Service) generateAlias() string {
 	alias := strings.Replace(s.Name, " ", "-", -1)
 	alias = strings.Trim(alias, " _-.,")
 	alias = strings.ToLower(alias)
-	return alias
+	return alias + s.computeHash()
 }
diff --git a/service/service.go b/service/service.go
index 15701cc3..1e44fb1c 100644
--- a/service/service.go
+++ b/service/service.go
@@ -207,6 +207,10 @@ func (s *Service) deploy() error {
 	s.configuration.Key = "service"
 	s.configuration.Image = imageHash
 	s.Dependencies = append(s.Dependencies, s.configuration)
+
+	if s.Alias == "" {
+		s.Alias = s.generateAlias()
+	}
 	return nil
 }
```

Reply:
Ok I change the behavior on empty Alias like you suggest.
One little dif, the alias is directly equal to the service hash:
```go
if s.Alias == "" {
    s.Alias = s.computeHash()
}
```